src/coreclr/vm/amd64/CallDescrWorkerAMD64.asm

```diff
@@ -42,6 +42,11 @@ StackCopyLoop:                          ; copy the arguments to stack top-down t
 
         mov     rax, [rbx + CallDescrData__dwRegTypeMap] ; save the reg (arg) type map
 
+        ; Store stub context
+        cmp     QWORD PTR [rbx + CallDescrData__pStubContextMD], 0
+        jz      NoStubContext
```

Comment:
How do we know that `r10` is volatile in this case? I'm hoping we can simply avoid stub sharing in interpreter scenarios.

Reply:
The r10 is volatile in both the windows and unix x64 calling convention. And this function is only called from native runtime code, so there is no case when a special calling convention extension that we use for some stubs could expect r10 to be preserved.

Reply:
Oh! You mean volatile in the calling convention sense, not `volatile` in the memory sense. Yes, I think we can unconditionally play with `r10`. I added the check just in case it was needed by another scenario that I didn't know about.